src/jit/lsraxarch.cpp

```diff
@@ -832,31 +896,28 @@ bool LinearScan::isRMWRegOper(GenTree* tree)
 //
 int LinearScan::BuildShiftRotate(GenTree* tree)
 {
-    TreeNodeInfo* info = currentNodeInfo;
     // For shift operations, we need that the number
     // of bits moved gets stored in CL in case
     // the number of bits to shift is not a constant.
-    int                   srcCount    = 0;
-    GenTree*              shiftBy     = tree->gtOp.gtOp2;
-    GenTree*              source      = tree->gtOp.gtOp1;
-    LocationInfoListNode* shiftByInfo = nullptr;
+    int       srcCount      = 0;
+    GenTree*  shiftBy       = tree->gtOp.gtOp2;
+    GenTree*  source        = tree->gtOp.gtOp1;
+    regMaskTP srcCandidates = RBM_NONE;
+    regMaskTP dstCandidates = RBM_NONE;
+
     // x64 can encode 8 bits of shift and it will use 5 or 6. (the others are masked off)
     // We will allow whatever can be encoded - hope you know what you are doing.
     if (shiftBy->isContained())
     {
-        srcCount += GetOperandInfo(source);
+        assert(shiftBy->OperIsConst());
     }
     else
     {
-        srcCount++;
-        shiftByInfo = getLocationInfo(shiftBy);
-        shiftByInfo->info.setSrcCandidates(this, RBM_RCX);
-        info->setDstCandidates(this, allRegs(TYP_INT) & ~RBM_RCX);
-        LocationInfoListNode* sourceInfo;
-        srcCount += GetOperandInfo(source, &sourceInfo);
-        for (; sourceInfo != nullptr; sourceInfo = sourceInfo->Next())
+        srcCandidates = allRegs(TYP_INT) & ~RBM_RCX;
+        dstCandidates = allRegs(TYP_INT) & ~RBM_RCX;
+        if (tree->IsReverseOp())
```

Comment:
Reverse op? Wasn't this removed from LIR?

Reply:
Well, I feel a little silly - I had either forgotten or had missed that this happened. There's code in LSRA that attempts to handle it, but I see that the Rationalizer is clearing it on all nodes, and `CheckLIR` is validating it. I added an issue #16528